tests/machine.test.js

```diff
@@ -911,4 +913,121 @@ describe(OlmMachine.name, () => {
             });
         });
     });
+
+    describe("verifyBackup", () => {
+        test("rejects backups with unknown signature", async () => {
+            let m = await machine();
+
+            let backupData = {
+                version: "2",
+                algorithm: "m.megolm_backup.v1.curve25519-aes-sha2",
+                auth_data: {
+                    public_key: "ddIQtIjfCzfR69I/imE7XiGsPPKA1KF74aclXsiWh08",
+                    signatures: {
+                        "@web:example.org": {
+                            "ed25519:WVJSAIOBUZ":
+                                "zzqyWl3ek5dSWKKeNPrpMFDQyu9ZlHrA2XpAaXtcSyo8BoZIu0K2flfT+N0YgVee2gmAZdLAribwgoCopvTeAg",
+                            "ed25519:LHMKRoMYl7haWnst5Xo54DuRqjZ5h/Sk1lxc4heSEcI":
+                                "YwRj5UqKrbMbAb/VK0Dwj4HspiOjSN64cM5SwFQ7HEcFiHp4gJmHtV90kl+12OLiE5JqRWvgzsx61hSXM/JDCA",
+                        },
+                    },
+                },
+                etag: "0",
+                count: 0,
+            };
+
+            const state = await m.verifyBackup(backupData);
+
+            expect(state.deviceState).toStrictEqual(SignatureState.Missing);
+            expect(state.userState).toStrictEqual(SignatureState.Missing);
+        });
+
+        test("accepts own signatures", async () => {
+            let m = await machine();
+            let _ = m.bootstrapCrossSigning(true);
+
+            let keyBackupKey = BackupDecryptionKey.createRandomKey();
+
+            let auth_data = {
+                public_key: keyBackupKey.megolmV1PublicKey.publicKeyBase64,
```

Comment:
It's a bit confusing that the "decryption" key also contains the encryption key.

Case in point: the naming of `BackupDecryptionKey.createRandomKey()` suggests that it creates a random decryption key, when it actually creates a random public+private key pair.

Maybe `BackupDecryptionKey` (here & in [the PR from `matrix-rust-sdk`](https://github.com/matrix-org/matrix-rust-sdk/pull/2305)) should have been named just `BackupKey`?

Or this could be all moot if it's understood that generating a private key must also generate a public one to go along with it.

Reply:
Yeah, it's not super-clear, I agree.

It's less that the decryption key *contains* the encryption key, and more that the encryption key can be *derived* from the decryption key, which is just the way that curve25519 keys work. `megolmV1PublicKey` here is actually a getter function which does some light maths to derive the public key.

so...

> Case in point: the naming of BackupDecryptionKey.createRandomKey() suggests that it creates a random decryption key, when it actually creates a random public+private key pair.

well no, it does actually create a random (private) decryption key. The public key is then derived later on.

Reply:
Alright, let's leave it as it is then.